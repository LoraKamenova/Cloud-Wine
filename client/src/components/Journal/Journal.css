.journal-page {
    background-image: url('https://res.cloudinary.com/dyhmxus4n/image/upload/v1615655025/React%20Project/writing_z4lxds.jpg');
    height: 100vh;
}

.journal-section {
    margin-right: 7%;
}

.journal-page > .general-pages-wrapper > .journal-section {
    height: 480px;
    margin-top: 60px;
}

.journal-section > a {
    text-decoration: none;
}
