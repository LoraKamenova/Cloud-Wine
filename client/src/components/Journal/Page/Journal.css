.journal-image-wrapper {
    background-image: url('https://res.cloudinary.com/dyhmxus4n/image/upload/v1615990671/React%20Project/write_tjtfgj.jpg');
}

.journal-section > .journal-content-container > .journal-article-wrapper > .info-button-wrapper > a {
    text-decoration: none;
}


@media all and (max-width: 1200px) {
    .journal-image-container {
        display: none;
    }
    .journal-article-wrapper {
        width: 100%;
        height: auto;
        padding: 55px;
    }

    .info-button-wrapper {
        justify-content: center;
        margin-top: 20px;
    }
}
