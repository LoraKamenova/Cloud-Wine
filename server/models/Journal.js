const mongoose = require('mongoose');
const { String, ObjectId } = mongoose.Schema.Types;

const JournalSchema = new mongoose.Schema({
    title: {
        type: String,
        required: true,
    },
    description: {
        type: String,
        required: true,
    },
    imageUrl1: {
        type: String,
        required: true
    },
    imageUrl2:{
        type: String,
        required: true
    },
    creatorId: {
        type: ObjectId,
        ref: 'User' },
});

module.exports = mongoose.model('Journal', JournalSchema);
