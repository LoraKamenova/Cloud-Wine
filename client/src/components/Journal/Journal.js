import './Journal.css'
import Article from "../Article/Article";
import {Link} from "react-router-dom";

const Journal = () => {
    return (
        <section className="custom-section journal-section">
            <div className="journal-content-container">
                <div className="journal-article-wrapper">
                    <Article
                        title="ВИНОПИСИ"
                        paragraph="Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                            Phasellus pulvinar tellus nec ante volutpat, vel commodo nisl aliquam. Maecenas nec
                            malesuada risus, ut maximus libero. Aliquam bibendum sollicitudin nibh, ac vehicula leo
                            maximus vitae. Aenean vitae metus luctus, molestie leo ac, ullamcorper justo. Class aptent
                            taciti sociosqu ad litora torquent per conubia nostra, per"
                    />
                    <div className="info-button-wrapper">
                        <Link to="/all-journals"><span className="button info-button">Виж винописите<i
                            className="fas fa-angle-double-right"></i></span></Link>
                    </div>
                </div>
                <div className="journal-image-container">
                    <div className="journal-image-wrapper"></div>
                </div>
            </div>
        </section>
    );
};

export default Journal;
