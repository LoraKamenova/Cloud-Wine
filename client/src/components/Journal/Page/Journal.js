import './Journal.css'
import AsideArticle from "../../Shared/AsideArticle/AsideArticle";

import { Link } from "react-router-dom";

const Journal = () => {
    return (
        <section className="journal-section">
            <div className="journal-content-container">
                <div className="journal-article-wrapper">
                    <AsideArticle
                        title="ПЪТЕПИСИ"
                        paragraph="Тук е мястото, където ще събираме вашите пътеписи. Разкажете ни вашите положителни или отрицателни впечатления от посетената дестинация. Препоръчайте или не вината, които сте опитали. Покажените снимките, които сте заснели. Прочетете преживяванията на другите любители на виното."
                    />
                    <div className="info-button-wrapper">
                        <Link to="/journal/all"><span className="button info-button">Виж пътеписите<i className="fas fa-angle-double-right"></i></span></Link>
                    </div>
                </div>
                <div className="journal-image-container">
                    <div className="journal-image-wrapper"></div>
                </div>
            </div>
        </section>
    );
};

export default Journal;
