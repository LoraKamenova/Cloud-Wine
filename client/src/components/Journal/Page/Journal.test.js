import { render } from '@testing-library/react';
import Journal from './Journal';
import {BrowserRouter} from "react-router-dom";

describe('Journal Component', () => {

    it("Journal Component snapshot", () => {
        const { asFragment } = render(<BrowserRouter><Journal /></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot();
    });

});
