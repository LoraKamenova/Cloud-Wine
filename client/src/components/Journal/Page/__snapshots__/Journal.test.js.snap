// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Journal Component Journal Component snapshot 1`] = `
<DocumentFragment>
  <section
    class="journal-section"
  >
    <div
      class="journal-content-container"
    >
      <div
        class="journal-article-wrapper"
      >
        <article
          class="article"
        >
          <div
            class="article-heading-wrapper"
          >
            <span
              class="article-heading"
            >
              ПЪТЕПИСИ
            </span>
          </div>
          <p
            class="article-paragraph"
          >
            Тук е мястото, където ще събираме вашите пътеписи. Разкажете ни вашите положителни или отрицателни впечатления от посетената дестинация. Препоръчайте или не вината, които сте опитали. Покажените снимките, които сте заснели. Прочетете преживяванията на другите любители на виното.
          </p>
        </article>
        <div
          class="info-button-wrapper"
        >
          <a
            href="/journal/all"
          >
            <span
              class="button info-button"
            >
              Виж пътеписите
              <i
                class="fas fa-angle-double-right"
              />
            </span>
          </a>
        </div>
      </div>
      <div
        class="journal-image-container"
      >
        <div
          class="journal-image-wrapper"
        />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
